app/assets/javascripts/darkswarm/services/authentication_service.js.coffee

```diff
@@ -32,6 +33,8 @@ Darkswarm.factory "AuthenticationService", (Navigation, $modal, $location, Redir
         'registration_authentication.html'
       else
         'authentication.html'
+    pushMessage: (message) ->
+      @modalMessage = String(message)
```

Comment:
it's a bit odd for me to violate the encapsulation of this class. I haven't dug deeper into what this service does but having to pass an error message to a service to be displayed it's a bit strange. It feels likes this service might too many responsibilities. Does it deal with the details of the authentication or simply encapsulates a modal or some other UI component :thinking: ?

Reply:
In this case, the checkout controller needed to open the login modal on a specific tab (which this service allows), but also display a specific message related to the checkout process inside the modal at the same time. The message is passed from the checkout controller via this service, to the individual controller for the specific tab.

Reply:
It's clearly not the scope of this PR but according to what you said this class has way too many responsibilities and we would better off with a better name for the class itself.